core/src/avm2/array.rs

```diff
@@ -85,44 +299,132 @@ impl<'gc> ArrayStorage<'gc> {
     ///
     /// Holes are copied as holes and not resolved at append time.
     pub fn append(&mut self, other_array: &Self) {
-        for other_item in other_array.storage.iter() {
-            self.storage.push(*other_item)
+        match self {
+            ArrayStorage::Dense(storage, dense_used) => match &other_array {
+                ArrayStorage::Dense(other_storage, _other_dense_used) => {
+                    for other_item in other_storage.iter() {
+                        storage.push(*other_item)
+                    }
+                    *dense_used = storage.iter().filter(|v| v.is_some()).count();
```

Comment:
Can't you just add `dense_used` and `other_dense_used`?

Reply:
I didn't realize that, thanks.